diff --git a/risc0/sys/kernels/zkp/cuda/ffi.cu b/risc0/sys/kernels/zkp/cuda/ffi.cu
--- a/risc0/sys/kernels/zkp/cuda/ffi.cu
+++ b/risc0/sys/kernels/zkp/cuda/ffi.cu
@@ -136,6 +136,8 @@
     CudaStream stream;
     combos_prepare<<<1, 1, 0, stream>>>(
         combos, coeffU, regsCount, regSizes, regComboIds, cycles, mix, checkSize, comboCount);
+    CUDA_OK(cudaGetLastError());
+    CUDA_OK(cudaStreamSynchronize(stream));
   } catch (const std::exception& err) {
     return strdup(err.what());
   }
